Import error while trying to parse irrelevant command line arguments

<!--
**NOTE** Please use this template to open issues, bugs, etc., only.
See our [GitHub Discussions Board](https://github.com/datapane/datapane/discussions) to discuss feature requests, general support, ideas, and to chat with the community.
-->

### System Information

<!-- Please fill this out to help us understand the bug/issue -->

- OS: Mac OS
- Python version: 3.7.7
- Python environment: pip
- Using jupyter: false
- Datapane version: 0.12.3

### Bug / Issue
Hi there,  we recently started to use datapane on our team to generate reports and we really like it.

We encountered an issue in our CI.
Whenever we import datapane in one of our tests, we get the following error: 
`click.exceptions.BadParameter: 'no:warnings', should be name=value`

The reason is we run our tests using:
`python -m pytest tests/ -p no:warnings`

But datapane in its main __init__.py module tries to parse any command line parameter starting with '-p', and in our case fails because this flag is for pytest's use.

I'm not sure what the correct behavior should be, one option is to disable this override mechanism when datapane is used as a lib (and not an explicit datapane command), another option is to use more specific flag prefix instead of just '-p', maybe '--datapane-p' or something similar.

Hope this issue is not a duplicate, I didn't find anything similar.

Full error stack:
```
/usr/local/lib/python3.7/site-packages/datapane/client/utils.py:104: in split_param
    k, v = x.split("=", maxsplit=1)
E   ValueError: not enough values to unpack (expected 2, got 1)
During handling of the above exception, another exception occurred:
tests/reporting/test_report_creator.py:8: in <module>
    from reporting.serve_report.report_creator import _get_report_pages_block
reporting/serve_report/report_creator.py:4: in <module>
    import datapane as dp
/usr/local/lib/python3.7/site-packages/datapane/__init__.py:87: in <module>
    config = parse_command_line()
/usr/local/lib/python3.7/site-packages/datapane/client/utils.py:132: in parse_command_line
    return process_cmd_param_vals(dp_args.parameter)
/usr/local/lib/python3.7/site-packages/datapane/client/utils.py:111: in process_cmd_param_vals
    return {k: convert_param_val(v) for (k, v) in split_param(params)}
/usr/local/lib/python3.7/site-packages/datapane/client/utils.py:111: in <dictcomp>
    return {k: convert_param_val(v) for (k, v) in split_param(params)}
/usr/local/lib/python3.7/site-packages/datapane/client/utils.py:106: in split_param
    raise click.BadParameter(err_msg.format(x))
E   click.exceptions.BadParameter: 'no:warnings', should be name=value
```


<!--
A clear and concise description of the issue.

Add as much info as needed, including steps to reproduce, expected behaviour, and even screenshots if helps.
-->

I've reproduced this with a simple test:

```python
def test_datapane_report():
    import datapane as dp
    import os

    report = dp.Report("# My report")
    report.save(path="my_report.html")
    assert os.path.exists("my_report.html") == False
```


```
shahin@datapane ~/Documents/reproduce_182 % poetry run python3 -m pytest tests/ -p no:warnings
================================================================================== test session starts ==================================================================================
platform darwin -- Python 3.10.5, pytest-7.1.3, pluggy-0.13.1
rootdir: /Users/shahin/Documents/reproduce_182
collected 2 items                                                                                                                                                                       

tests/test_reproduce_182.py .F                                                                                                                                                    [100%]

======================================================================================= FAILURES ========================================================================================
_________________________________________________________________________________ test_datapane_report __________________________________________________________________________________

xs = ['no:warnings']

    def split_param(xs: Tuple[str, ...]) -> Iterator[t.Tuple[str, str]]:
        err_msg = "'{}', should be name=value"
        for x in xs:
            try:
>               k, v = x.split("=", maxsplit=1)
E               ValueError: not enough values to unpack (expected 2, got 1)

.venv/lib/python3.10/site-packages/datapane/client/utils.py:125: ValueError

During handling of the above exception, another exception occurred:

    def test_datapane_report():
>       import datapane as dp

tests/test_reproduce_182.py:9: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/python3.10/site-packages/datapane/__init__.py:132: in <module>
    config = parse_command_line()
.venv/lib/python3.10/site-packages/datapane/client/utils.py:153: in parse_command_line
    return process_cmd_param_vals(dp_args.parameter)
.venv/lib/python3.10/site-packages/datapane/client/utils.py:132: in process_cmd_param_vals
    return {k: convert_param_val(v) for (k, v) in split_param(params)}
.venv/lib/python3.10/site-packages/datapane/client/utils.py:132: in <dictcomp>
    return {k: convert_param_val(v) for (k, v) in split_param(params)}
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

xs = ['no:warnings']

    def split_param(xs: Tuple[str, ...]) -> Iterator[t.Tuple[str, str]]:
        err_msg = "'{}', should be name=value"
        for x in xs:
            try:
                k, v = x.split("=", maxsplit=1)
            except ValueError:
>               raise click.BadParameter(err_msg.format(x))
E               click.exceptions.BadParameter: 'no:warnings', should be name=value

.venv/lib/python3.10/site-packages/datapane/client/utils.py:127: BadParameter
================================================================================ short test summary info ================================================================================
FAILED tests/test_reproduce_182.py::test_datapane_report - click.exceptions.BadParameter: 'no:warnings', should be name=value
============================================================================== 1 failed, 1 passed in 8.18s ==============================================================================
```

--- a/projects/python-client/src/datapane/client/commands.py
+++ b/projects/python-client/src/datapane/client/commands.py
@@ -320,7 +320,7 @@ def app_list():
 
 
 @app.command()
-@click.option("--parameter", "-p", multiple=True)
+@click.option("--dp-parameter", multiple=True)
 @click.option("--cache/--disable-cache", default=True)
 @click.option("--wait/--no-wait", default=True)
 @click.option("--project")
@@ -328,14 +328,14 @@ def app_list():
 @click.argument("name")
 def run(
     name: str,
-    parameter: Tuple[str],
+    dp_parameter: Tuple[str],
     cache: bool,
     wait: bool,
     project: str,
     show_output: bool,
 ):
     """Run a report"""
-    params = process_cmd_param_vals(parameter)
+    params = process_cmd_param_vals(dp_parameter)
     log.info(f"Running app with parameters {params}")
     app = api.App.get(name, project=project)
     with api_error_handler("Error running app"):
@@ -492,20 +492,20 @@ def schedule():
 
 
 @schedule.command()  # type: ignore[no-redef]
-@click.option("--parameter", "-p", multiple=True)
+@click.option("--dp-parameter", multiple=True)
 @click.argument("name", required=True)
 @click.argument("cron", required=True)
 @click.option("--project")
-def create(name: str, cron: str, parameter: Tuple[str], project: str):
+def create(name: str, cron: str, dp_parameter: Tuple[str], project: str):
     """
     Create a schedule
 
     NAME: Name of the App to run
     CRON: crontab representing the schedule interval
-    PARAMETERS: key/value list of parameters to use when running the app on schedule
+    DP_PARAMETERS: key/value list of parameters to use when running the app on schedule
     [Project]: Project in which the app is present
     """
-    params = process_cmd_param_vals(parameter)
+    params = process_cmd_param_vals(dp_parameter)
     log.info(f"Adding schedule with parameters {params}")
     app_obj = api.App.get(name, project=project)
     schedule_obj = api.Schedule.create(app_obj, cron, params)
@@ -513,20 +513,20 @@ def create(name: str, cron: str, parameter: Tuple[str], project: str):
 
 
 @schedule.command()
-@click.option("--parameter", "-p", multiple=True)
+@click.option("--dp-parameter", multiple=True)
 @click.argument("id", required=True)
 @click.argument("cron", required=False)
-def update(id: str, cron: str, parameter: Tuple[str]):
+def update(id: str, cron: str, dp_parameter: Tuple[str]):
     """
     Add a schedule
 
     ID: ID/URL of the Schedule
     CRON: crontab representing the schedule interval
-    PARAMETERS: key/value list of parameters to use when running the app on schedule
+    DP_PARAMETERS: key/value list of parameters to use when running the app on schedule
     """
 
-    params = process_cmd_param_vals(parameter)
-    assert cron or parameter, "Must update either cron or parameters"
+    params = process_cmd_param_vals(dp_parameter)
+    assert cron or dp_parameter, "Must update either cron or parameters"
 
     log.info(f"Updating schedule with parameters {params}")
 

--- a/projects/python-client/src/datapane/client/utils.py
+++ b/projects/python-client/src/datapane/client/utils.py
@@ -134,10 +134,13 @@ def process_cmd_param_vals(params: Tuple[str, ...]) -> JDict:
 
 def parse_command_line() -> t.Dict[str, t.Any]:
     """Called in library mode to pull any parameters into dp.Config"""
-    parser = argparse.ArgumentParser(description="Datapane additional args", conflict_handler="resolve", add_help=False)
+    parser = argparse.ArgumentParser(
+        description="Datapane additional args",
+        conflict_handler="resolve",
+        add_help=False,
+    )
     parser.add_argument(
-        "--parameter",
-        "-p",
+        "--dp-parameter",
         action="append",
         help="key=value parameters to pass into dp.Params",
         default=[],
@@ -150,4 +153,4 @@ def parse_command_line() -> t.Dict[str, t.Any]:
     sys.argv.append(exe_name)
     sys.argv.extend(remaining_args)
 
-    return process_cmd_param_vals(dp_args.parameter)
+    return process_cmd_param_vals(dp_args.dp_parameter)

--- a/projects/python-client/tests/client/e2e/test_library.py
+++ b/projects/python-client/tests/client/e2e/test_library.py
@@ -13,7 +13,7 @@ def test_arbitrary_arguments_from_cli():
         "pytest",
         test_file_path,
         "-k",
-        "test_pytest_from_cli",
+        "test_datapane_init_from_cli",
         "-p",
         "no:warnings",
     ]
@@ -32,7 +32,7 @@ def test_no_arguments_from_cli():
         "pytest",
         test_file_path,
         "-k",
-        "test_pytest_from_cli",
+        "test_datapane_init_from_cli",
     ]
 
     p = subprocess.run(test_command)
@@ -40,7 +40,7 @@ def test_no_arguments_from_cli():
     assert p.returncode == 0
 
 
-def test_pytest_from_cli():
+def test_datapane_init_from_cli():
     """This test is run by keyword expression from other CLI tests.
     It explicitly imports datapane to ensure __init__ is always executed."""
     import datapane as dp

--- a/projects/python-client/tests/client/e2e/test_library.py
+++ b/projects/python-client/tests/client/e2e/test_library.py
@@ -43,6 +43,6 @@ def test_no_arguments_from_cli():
 def test_datapane_init_from_cli():
     """This test is run by keyword expression from other CLI tests.
     It explicitly imports datapane to ensure __init__ is always executed."""
-    import datapane as dp
+    import datapane as dp   # noqa: F401
 
     assert True

--- a/projects/python-client/tests/client/e2e/test_library.py
+++ b/projects/python-client/tests/client/e2e/test_library.py
@@ -1,0 +1,48 @@
+import subprocess
+import sys
+import os
+
+
+def test_arbitrary_arguments_from_cli():
+    """Running the pytest command with arguments for -p
+    to make sure Datapane doesn't process them."""
+    test_file_path = os.path.realpath(__file__)
+    test_command = [
+        sys.executable,
+        "-m",
+        "pytest",
+        test_file_path,
+        "-k",
+        "test_pytest_from_cli",
+        "-p",
+        "no:warnings",
+    ]
+
+    p = subprocess.run(test_command, capture_output=True)
+
+    assert p.returncode == 0, str(p.stderr).replace("\\n", "\n")
+
+
+def test_no_arguments_from_cli():
+    """Running the pytest command without arguments for -p."""
+    test_file_path = os.path.realpath(__file__)
+    test_command = [
+        sys.executable,
+        "-m",
+        "pytest",
+        test_file_path,
+        "-k",
+        "test_pytest_from_cli",
+    ]
+
+    p = subprocess.run(test_command)
+
+    assert p.returncode == 0
+
+
+def test_pytest_from_cli():
+    """This test is run by keyword expression from other CLI tests.
+    It explicitly imports datapane to ensure __init__ is always executed."""
+    import datapane as dp
+
+    assert True

--- a/projects/python-client/tests/client/e2e/test_library.py
+++ b/projects/python-client/tests/client/e2e/test_library.py
@@ -43,6 +43,6 @@ def test_no_arguments_from_cli():
 def test_datapane_init_from_cli():
     """This test is run by keyword expression from other CLI tests.
     It explicitly imports datapane to ensure __init__ is always executed."""
-    import datapane as dp   # noqa: F401
+    import datapane as dp  # noqa: F401
 
     assert True

--- a/projects/python-client/tests/client/e2e/test_library.py
+++ b/projects/python-client/tests/client/e2e/test_library.py
@@ -3,7 +3,7 @@ import subprocess
 import sys
 
 
-def test_arbitrary_arguments_from_cli():
+def test_pytest_arguments_from_cli():
     """Running the pytest command with arguments for -p
     to make sure Datapane doesn't process them."""
     test_file_path = os.path.realpath(__file__)
@@ -23,7 +23,7 @@ def test_arbitrary_arguments_from_cli():
     assert p.returncode == 0, str(p.stderr).replace("\\n", "\n")
 
 
-def test_no_arguments_from_cli():
+def test_no_pytest_arguments_from_cli():
     """Running the pytest command without arguments for -p."""
     test_file_path = os.path.realpath(__file__)
     test_command = [
@@ -33,6 +33,55 @@ def test_no_arguments_from_cli():
         test_file_path,
         "-k",
         "test_datapane_init_from_cli",
+    ]
+
+    p = subprocess.run(test_command)
+
+    assert p.returncode == 0
+
+
+def test_correctly_formatted_dp_parameter_arguments_from_cli():
+    """Running a Python program with correctly formatted arguments for
+    --dp-parameter to make sure Datapane doesn't fail when imported."""
+    test_command = [
+        sys.executable,
+        "-c",
+        "import datapane",
+        "--dp-parameter",
+        "my_param=arg",
+    ]
+
+    p = subprocess.run(test_command)
+
+    assert p.returncode == 0
+
+
+def test_incorrectly_formatted_dp_parameter_arguments_from_cli():
+    """Running a Python program with incorrectly formatted arguments for
+    --dp-parameter to make sure Datapane fails when imported."""
+    test_command = [
+        sys.executable,
+        "-c",
+        "import datapane",
+        "--dp-parameter",
+        "my_param:arg",
+    ]
+
+    p = subprocess.run(test_command)
+
+    # Should fail, we're passing in badly formatted args to --dp-parameter
+    assert p.returncode != 0
+
+
+def test_arbitrary_arguments_from_cli():
+    """Running a Python program with arbitrary arguments for -arbs
+    to make sure Datapane doesn't process them when imported."""
+    test_command = [
+        sys.executable,
+        "-c",
+        "import datapane",
+        "-arbs",
+        "my_param:arg",
     ]
 
     p = subprocess.run(test_command)

--- a/projects/python-client/tests/client/e2e/test_library.py
+++ b/projects/python-client/tests/client/e2e/test_library.py
@@ -1,6 +1,6 @@
+import os
 import subprocess
 import sys
-import os
 
 
 def test_arbitrary_arguments_from_cli():

